src/transformers/generation/utils.py

```diff
@@ -1399,6 +1403,19 @@ def generate(
                     "(https://huggingface.co/docs/transformers/main/en/main_classes/text_generation)"
                 )
             generation_config.max_length = generation_config.max_new_tokens + input_ids_length
+
+        # if we don't pass `past_key_values` and a cache_implementation is specified
+        if generation_config.cache_implementation in ALL_COMPILE_CACHE_CLASSES_MAPPING and not model_kwargs.get(
```

Comment:
```suggestion
        if generation_config.cache_implementation in NEED_SETUP_CACHE_CLASSES_MAPPING and not model_kwargs.get(
```
nit: let's use a more flexible naming :) (compilable caches need setup, but the reverse may not be true)